src/libs/actions/Task.js

```diff
@@ -0,0 +1,158 @@
+    const optimisticTaskReport = ReportUtils.buildOptimisticTaskReport(currentUserEmail, assignee, parentReportID, optimisticCreatedAction.reportActionID, name, description);
+
+    // AddCommentReportAction on the parent chat report
+    const AddCommentText = ` created a task: ${name}`;
+    const optimisticAddCommentReportAction = ReportUtils.buildOptimisticAddCommentReportAction(AddCommentText);
+
+    // Open the parent report if assignee is given
+    if (assignee) {
+        Report.openReport(parentReportID, [assignee]);
+    }
+
+    const optimisticData = [
+        {
+            method: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
+            value: {[optimisticCreatedAction.reportActionID]: optimisticCreatedAction},
+        },
+
+        // {
+        //     method: CONST.ONYX.METHOD.MERGE,
+        //     key: `${ONYXKEYS.COLLECTION.REPORT}${parentReportID}`,
+        //     value: parentReport,
+        // },
+        {
+            method: CONST.ONYX.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: optimisticTaskReport,
+        },
+        optimisticAddCommentReportAction,
+    ];
+
+    const successData = [];
+
+    const failureData = [
+        {
+            method: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
+            value: {[optimisticCreatedAction.reportActionID]: {pendingAction: null}},
+        },
+
+        // {
+        //     method: CONST.ONYX.METHOD.MERGE,
+        //     key: `${ONYXKEYS.COLLECTION.REPORT}${parentReportID}`,
+        //     value: null,
+        // },
+        {
+            method: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: null,
+        },
+    ];
+
+    API.write(
+        'CreateTask',
+        {
+            parentReportActionID: optimisticCreatedAction.reportActionID,
+            parentReportID,
+            taskReportID: optimisticTaskReport.reportID,
+            name,
+            description,
+            assignee,
+            assigneeChatReportID,
+        },
+        {optimisticData, successData, failureData},
+    );
+
+    clearOutTaskInfo();
+
+    // TODO: Navigate to the newly created task
+    Navigation.navigate(ROUTES.getReportRoute(parentReportID));
+}
+
+function setDetailsValue(name, description) {
+    // This is only needed for creation of a new task and so it should only be stored locally
+    Onyx.merge(ONYXKEYS.TASK, {name, description});
+}
+
+function setTitleValue(name) {
+    Onyx.merge(ONYXKEYS.TASK, {name});
+}
+
+function setDescriptionValue(description) {
+    Onyx.merge(ONYXKEYS.TASK, {description});
+}
+
+function setAssigneeValue(assignee) {
+    // This is only needed for creation of a new task and so it should only be stored locally
+    Onyx.merge(ONYXKEYS.TASK, {assignee});
+}
```

Comment:
Here is where we'd need to add the check

Do we already have a share destination set? 
No - optimistically create a DM with the user if one does not already exist
And set it as the shareDestination

Yes - do nothing because we will share it in the already existing room/chat/gc

Reply:
So in this scenario, I actually don't think we need to pass the optimistic chat report DM because we will create it before we get to that point